File: projects/repo-intake-fbabaei-microsoft-azure-architecture-factory-20260425012922/projects/mdr-support-20260421003611/src/Services/ClarificationService.cs
namespace GeneratedApi.Services;

using GeneratedApi.Models;

public class ClarificationService
{
    private static readonly Dictionary<string, string> FieldPrompts = new()
    {
        ["reference_id"] = "What is the reference ID for this submission?",
        ["submission_date"] = "What is the submission date (YYYY-MM-DD)?",
        ["jurisdiction"] = "Which jurisdiction does this arrangement apply to?",
        ["summary"] = "Please provide a short summary of the arrangement.",
    };

    public List<Clarification> ComputeMissingFields(ExtractionDraft draft)
    {
        var missing = new List<Clarification>();
        foreach (var field in ExtractionService.MandatoryFields)
        {
            draft.Fields.TryGetValue(field, out var value);
            if (string.IsNullOrWhiteSpace(value))
            {
                var prompt = FieldPrompts.TryGetValue(field, out var p)
                    ? p
                    : $"Please provide a value for '{field}'.";
                missing.Add(new Clarification(field, prompt));
            }
        }
        return missing;
    }
}

File: projects/repo-intake-fbabaei-microsoft-azure-architecture-factory-20260425012922/projects/mdr-support-20260421002041/src/mdr_support/services/clarification_service.py
"""Clarification service: compute the next missing mandatory field."""
from __future__ import annotations

from ..models import Clarification, ExtractionDraft
from .extraction_service import MANDATORY_FIELDS


_FIELD_PROMPTS: dict[str, str] = {
    "reference_id": "What is the reference ID for this submission?",
    "submission_date": "What is the submission date (YYYY-MM-DD)?",
    "jurisdiction": "Which jurisdiction does this arrangement apply to?",
    "summary": "Please provide a short summary of the arrangement.",
}


def compute_missing_fields(draft: ExtractionDraft) -> list[Clarification]:
    missing: list[Clarification] = []
    for field in MANDATORY_FIELDS:
        value = draft.fields.get(field)
        if value is None or (isinstance(value, str) and not value.strip()):
            missing.append(Clarification(field=field, prompt=_FIELD_PROMPTS.get(field, "Please provide a value.")))
    return missing
